packages/react-components/react-menu/src/components/MenuPopover/useMenuPopoverStyles.styles.ts

```diff
@@ -15,6 +15,7 @@ const useStyles = makeStyles({
     color: tokens.colorNeutralForeground1,
     minWidth: '128px',
     maxWidth: '300px',
+    overflowX: 'hidden',
```

Comment:
MenuDivider has -5px margin left/right to make sure it piece through MenuPopover's padding/border.

But when MenuPopover has `overflow-y:auto`, css automatically sets overflow-x to auto as well. When MenuPopover is positioned with subpixels, MenuDivider can create 1px horizontal overflow on MenuPopover:
<img width="341" alt="Screenshot 2023-07-28 at 18 24 06" src="https://github.com/microsoft/fluentui/assets/28751745/dd4764ec-4ad5-4e69-be01-3888f91ae2c7">

Since Menu has maxWith and should never have horizontal scrolling, I'm adding this style here.